include/libdnf/utils/utils.hpp

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_UTILS_UTILS_HPP
+#define LIBDNF_UTILS_UTILS_HPP
+
+namespace libdnf {
+
+bool have_files_same_content(const char * file_path1, const char * file_path2);
```

Comment:
Please could you add an note that in case of any problem it returns false, like missing file, permissions or so on.

Reply:
In such cases it should throw an exception, not return `false`.

Reply:
How about following?
files_have_same_content_noexcept() - return true if they match, false otherwise (incl. after throwing any exception)

And add following in the future if needed:
files_have_same_content() - throwing an exception

Reply:
@dmach possibly, I haven't looked at the usage of this function, but unless the `_noexcept()` version saves a lot of try-catches **and we are sure** it won't hide any errors from the user (which I am quite sceptical about), I'd prefer to just have the try-catch around the use of this function which logs the error and handles it in the appropriate way. That simply is much cleaner and it tends to pay off in the long run...

Reply:
I marked function as `noexcept` and added `_noexcept` suffix to its name. The docs string was added too.